lib/main.dart

```diff
@@ -1,61 +1,84 @@
 import 'dart:async';
+
+import 'package:aedex/application/dex_token.dart';
+import 'package:aedex/application/pool/dex_pool.dart';
+import 'package:aedex/application/verified_tokens.dart';
 import 'package:aedex/infrastructure/hive/db_helper.hive.dart';
 import 'package:aedex/router/router.dart';
+import 'package:aedex/ui/views/farm_lock/bloc/provider.dart';
+import 'package:aedex/ui/views/pool_list/bloc/provider.dart';
 import 'package:aedex/util/service_locator.dart';
 import 'package:archethic_dapp_framework_flutter/archethic_dapp_framework_flutter.dart'
     as aedappfm;
 import 'package:flutter/material.dart';
 import 'package:flutter_gen/gen_l10n/localizations.dart';
 import 'package:flutter_localizations/flutter_localizations.dart';
-import 'package:flutter_native_splash/flutter_native_splash.dart';
 import 'package:flutter_riverpod/flutter_riverpod.dart';
 import 'package:go_router/go_router.dart';
 import 'package:url_strategy/url_strategy.dart';
 
 Future<void> main() async {
-  final widgetsBinding = WidgetsFlutterBinding.ensureInitialized();
   GoRouter.optionURLReflectsImperativeAPIs = true;
-  FlutterNativeSplash.preserve(widgetsBinding: widgetsBinding);
   await DBHelper.setupDatabase();
-  setupServiceLocator();
+  await setupServiceLocator();
+
   setPathUrlStrategy();
   runApp(
     ProviderScope(
       observers: [
         aedappfm.ProvidersLogger(),
       ],
-      child: const MyApp(),
+      child: const ProvidersInitialization(child: MyApp()),
     ),
   );
 }
 
+/// Eagerly initializes providers (https://riverpod.dev/docs/essentials/eager_initialization).
+///
+/// Add Watch here for any provider you want to init when app is displayed.
+/// Those providers will be kept alive during application lifetime.
+class ProvidersInitialization extends ConsumerWidget {
+  const ProvidersInitialization({required this.child, super.key});
+
+  final Widget child;
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    ref
```

Comment:
is the order is important to load info ?

Reply:
No, this is not important.